app/controllers/concerns/history_helper.rb

```diff
@@ -0,0 +1,64 @@
+module HistoryHelper
```

Comment:
If we made this a class, we could remove the need to set `@history`, or call `.reset`?

What do you think?